jwst/extract_1d/soss_extract/atoca.py

```diff
@@ -933,6 +933,48 @@ def __call__(self, data, error, tikhonov=False, factor=None):
 
         return spectrum
 
+
+    def return_matrices(self, data, error, tikfac):
```

Comment:
I think this name is a bit too generic.  I'd prefer something like `precompute_detector_model()` or whatever sounds good to you